File: packages/common/erp-common-ms-3-js/src/modules/country/country.controller.ts
import { ICountry, ICreateCountry, IUpdateCountry, IDeleteCountry, ISearchCountry } from '@gmahechas/erp-common';
import { Country } from './country.mongo';

export const createOneCountry = async (data: ICreateCountry): Promise<ICountry> => {
	const result = await Country.create(data);
	return result;
};

export const updateOneCountry = async (data: IUpdateCountry): Promise<ICountry | null> => {
	const result = await Country.findOneAndUpdate({ id: data.id }, data);
	return Object.assign(result, data);
};

export const deleteOneCountry = async (data: IDeleteCountry): Promise<ICountry | null> => {
	const result = await Country.findOneAndDelete(data);
	return result;
};

export const searchOneCountry = async (data: Partial<ISearchCountry>): Promise<ICountry | null> => {
	const result = await Country.findOne(data);
	return result;
};

export const searchManyCountry = async (data: Partial<ISearchCountry>[]): Promise<ICountry[]> => {
	const result = await Country.find({ $or: data });
	return result;
};

File: packages/3/ms-3-flaskpy/flaskr/routes/v1/country/country_controller.py
from .country_mongo import CountryMongo


class CountryController:

    country_mongo = CountryMongo()

    def create_one_country(self, data):
        result = self.country_mongo.create_one(data)
        return result

    def update_one_country(self, data):
        result = self.country_mongo.update_one(data)
        return result

    def delete_one_country(self, data):
        result = self.country_mongo.delete_one(data)
        return result

    def search_one_country(self, data):
        result = self.country_mongo.search_one(data)
        return result

    def search_many_country(self, data):
        result = self.country_mongo.search_many(data)
        return result
